src/services/serverService.ts

```diff
@@ -21,4 +21,15 @@ export class ServerService {
   public async getVersion(): Promise<Either<ActionResponse, string>> {
     return this.utils.get(this.endpoint, (data) => data.version);
   }
+
+  /**
+   * Get server report for a given nodeName
+   */
+  public downloadReport(nodeName: string): void {
```

Comment:
server errors are not handled there

Reply:
is this the way it works in Patternfly ?

Reply:
Changed function to handle server error. 
There's no mention of the download feature in patternfly.